test/integration/discordactions.test.js

```diff
@@ -248,4 +251,58 @@ describe("Discord actions", function () {
         });
     });
   });
+
+  describe("PUT /discord-actions/group-idle-7d", function () {
+    let allIds;
+    beforeEach(async function () {
+      userData[0].roles = { archived: false };
+      userData[1].roles = { archived: false };
+      userData[2].roles = { archived: false };
+      await addUser(userData[0]);
+      await addUser(userData[1]);
+      await addUser(userData[2]);
+
+      const addUsersPromises = userData.slice(0, 3).map((user) => userModel.add({ ...user }));
+      const responses = await Promise.all(addUsersPromises);
+      allIds = responses.map((response) => response.id);
+
+      const userStatusPromises = allIds.map(async (userId) => {
+        await updateUserStatus(userId, generateUserStatusData("IDLE", 1690829925336, 1690829925336));
+      });
+      await Promise.all(userStatusPromises);
+
+      const addRolesPromises = [discordRoleModel.add(groupIdle7d)];
+      await Promise.all(addRolesPromises);
+
+      fetchStub.returns(
+        Promise.resolve({
+          status: 200,
+          json: () => Promise.resolve(getDiscordMembers),
+        })
+      );
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+      await cleanDb();
+    });
+
+    it("should update Idle 7d+ Users successfully and return a 201 status code", function (done) {
+      chai
+        .request(app)
+        .put(`/discord-actions/group-idle-7d`)
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(201);
+          expect(res.body.message).to.be.equal("All Idle 7d+ Users updated successfully.");
+          expect(res.body.totalIdle7dUsers).to.be.equal(3);
+          expect(res.body.totalGroupIdle7dRolesApplied.count).to.be.equal(3);
+          expect(res.body.totalUserRoleToBeAdded).to.be.equal(3);
```

Comment:
Isn't it seems repetitive, as the same value **3** being checked for multiple cases ?
You can remove the last expect statement, if not needed

Reply:
it should show right how many users needs to add in group-idle-7d+ so i will keep it it's needed